week5/quine/neongreen/Main.hs

```diff
@@ -0,0 +1 @@
+main = putStrLn $ (++) <*> show $ "main = putStrLn $ (++) <*> show $ "
```

Comment:
Nice, this must be the,  or close to the, minimal solution!

Below I broke down roughly how I think the expressions reduce, so its clear how this works. The crux of the solution seems to be `(++) <*> show`. This is a rather lucky bit of logic to have at hand for this one-line solution : )


```haskell
putStrLn (((++) <*> show) "main = putStrLn $ (++) <*> show $ ")

putStrLn ("main = putStrLn $ (++) <*> show $ " ++ show "main = putStrLn $ (++) <*> show $ ")

putStrLn ("main = putStrLn $ (++) <*> show $ " ++ "\"main = putStrLn $ (++) <*> show $ \"")

putStrLn ("main = putStrLn $ (++) <*> show $ \"main = putStrLn $ (++) <*> show $ \"")
```
```
main = putStrLn $ (++) <*> show $ "main = putStrLn $ (++) <*> show $ "
```